src/gluonts/model/canonical/_estimator.py

```diff
@@ -68,31 +78,75 @@ def __init__(
         self.is_sequential = is_sequential
 
     def create_transformation(self) -> Transformation:
-        return Chain(
-            trans=[
-                AsNumpyArray(field=FieldName.TARGET, expected_ndim=1),
-                AddTimeFeatures(
-                    start_field=FieldName.START,
-                    target_field=FieldName.TARGET,
-                    output_field=FieldName.FEAT_TIME,
-                    time_features=time_features_from_frequency_str(self.freq),
-                    pred_length=self.prediction_length,
-                ),
-                SetFieldIfNotPresent(
-                    field=FieldName.FEAT_STATIC_CAT, value=[0.0]
-                ),
-                AsNumpyArray(field=FieldName.FEAT_STATIC_CAT, expected_ndim=1),
-                InstanceSplitter(
-                    target_field=FieldName.TARGET,
-                    is_pad_field=FieldName.IS_PAD,
-                    start_field=FieldName.START,
-                    forecast_start_field=FieldName.FORECAST_START,
-                    train_sampler=TestSplitSampler(),
-                    time_series_fields=[FieldName.FEAT_TIME],
-                    past_length=self.context_length,
-                    future_length=self.prediction_length,
-                ),
-            ]
+        return (
```

Comment:
This isn't so much related to the PR, but still:

When I look at this, a few things look odd.

Why is it called `create_transformation? What is transformed? Looking at the body, we do input pre-processing, ensuring that fields have the correct type and shape. Also we seem to add some features.

It looks very verbose and noisy, lots of `FieldName` being mentioned.

If we would introduce schemas, the code would already look much nicer.

Also, we might want to think about how we generate features.

Reply:
Yes, I agree. This is the same for several other estimators.